subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/pkg/algorithm/hived_algorithm.go

```diff
@@ -332,30 +410,121 @@ func (h *HivedAlgorithm) initFreeCellList() {
 func (h *HivedAlgorithm) initReservations() {
 	for vc, vcReservation := range h.reservedCells {
 		for rid, physical := range vcReservation {
+			h.vcRemainingQuotas[vc][physical.GetChain()][physical.GetLevel()]--
+			h.totalRemainingQuotas[physical.GetChain()][physical.GetLevel()]--
 			h.removeCellFromFreeList(physical)
 			virtualList := h.vcSchedulers[vc].getReservedCellList()[rid]
 			virtual := virtualList[CellLevel(len(virtualList))][0].(*VirtualCell)
-			virtual.SetPhysicalCell(physical)
-			physical.SetVirtualCell(virtual)
-			klog.Infof("Cells bound: %v and %v (reservation)", virtual.GetName(), physical.GetName())
+			bindCell(physical, virtual)
+		}
+	}
+}
+
+// setBad marks a physical cell as bad and recursively for its parent, guaranteeing that
+// a cell is bad if all of its children are bad. The virtual cell that this physical cell
+// is bound to will also be marked as bad. If the cell has not been bound to a virtual cell,
+// but the scheduler finds that a VC will be allocated such a bad cell,
```

Comment:
>allocated [](start = 45, length = 9)

"allocated" rename to "doomed", because it is not really allocated, let's distinguish them in all place. allocate for used, doomed to bad?